# HG changeset patch
# Parent 45c79bcee196ed153660a26f3998844c244ba522
LibJpeg: Replace Exit in error_exit with a function call that should raise an
exception.

diff -r 45c79bcee196 external/source/libjpeg/jerror.c
--- a/external/source/libjpeg/jerror.c	Fri Nov 15 15:03:21 2013 +0100
+++ b/external/source/libjpeg/jerror.c	Fri Nov 15 15:13:42 2013 +0100
@@ -66,6 +66,12 @@
  * or jpeg_destroy) at some point.
  */
 
+#define LIBTIFFDELPHI
+#ifdef LIBTIFFDELPHI
+/* LibTiffDelphi add jpeg_error_exit_raise, and implements it in LibJpegDelphi, to raise an exception the Delphi way */
+extern void jpeg_error_exit_raise(void);
+#endif
+
 METHODDEF(void)
 error_exit (j_common_ptr cinfo)
 {
@@ -75,7 +81,12 @@
   /* Let the memory manager delete any temp files before we die */
   jpeg_destroy(cinfo);
 
+#ifndef LIBTIFFDELPHI
   exit(EXIT_FAILURE);
+#else
+  /* LibTiffDelphi compatible exit */
+  jpeg_error_exit_raise();
+#endif 
 }
 
 
